Frontend/src/app/user-management/page.tsx

```diff
@@ -0,0 +1,275 @@
+"use client";
+import React, { useEffect, useRef, useState } from "react";
+import { useRouter } from "next/navigation";
+import { useDispatch } from "react-redux";
+import { ClusterData } from "@/data/cluster-data";
+import { IClusterDetails } from "@/types/cluster-types";
+import Highlighter from 'react-highlight-words';
+import { Tag, Button, Modal, Input, Space, Table, Layout, theme, Typography, message } from 'antd';
+import { SearchOutlined } from '@ant-design/icons';
+import type { SearchProps } from "antd/es/input/Search";
+import type { TableProps, PaginationProps } from 'antd';
+import type { InputRef, TableColumnType } from 'antd';
+import type { FilterDropdownProps } from 'antd/es/table/interface';
+import UserRegistrationForm from "@/components/cluster-details/user-registration-form";
+import PageWrapper from "@/layouts/page-wrapper";
+import { useAppSelector } from "@/redux/hook";
+import { IUserAccessData } from "@/types/user-types";
+import { getAllUsers } from "@/services/user-service";
+import { set_Users_Cache } from "@/redux/features/cacheSlice";
+
+const { Content } = Layout;
+
+interface DataType {
+  key: string;
+  userID: string;
+  Name: string;
+  Email: string;
+  Role: string;
+  Status: boolean;
+  [key: string]: string | boolean;
+}
+
+const PaginationProps = {
+  pageSize: 5,
+  defaultPageSize: 5,
+  showSizeChanger: true,
+  showTotal: (total: number, range: [number, number]) => `${range[0]}-${range[1]} of ${total} items`,
+} as PaginationProps;
+
+export default function Clusters() {
+  const router = useRouter();
+  const {
+    token: { colorBgContainer, borderRadiusLG },
+  } = theme.useToken();
+  const dispatch = useDispatch();
+  const { users, state } = useAppSelector(state => state.cacheData)
+  const [userData, setUserData] = useState<IUserAccessData[]>(users);
+  
+  const [openModal, setOpenModal] = useState<boolean>(false);
+
+  const [searchText, setSearchText] = useState('');
+  const [searchedColumn, setSearchedColumn] = useState('');
+  const searchInput = useRef<InputRef>(null);
+
+  // const onSearch: SearchProps["onSearch"] = (value, _e, info) => {
+  //   const filteredClusters = ClusterData.filter((cluster) => {
+  //     return cluster.name.toLowerCase().includes(value.toLowerCase());
+  //   });
+  //   setClusters(filteredClusters);
+  // }
+
+  const showModal = () => {
+    setOpenModal(true);
+  };
+
+  const handleOk = (e: React.MouseEvent<HTMLElement>) => {
+    setOpenModal(false);
+  };
+
+  const handleCancel = (e: React.MouseEvent<HTMLElement>) => {
+    setOpenModal(false);
+  };
+
+  const getTableData = () => {
+    return userData.map((data) => {
+      return {
+        key: data._id,
+        userID: data._id,
+        Name: data.fullName,
+        Email: data.email,
+        Role: data.role,
+        Status: data.enabled,
```

Comment:
```suggestion
        status: data.enabled,
```

Reply:
Is there a specific reason for lowercasing only the Status attribute?